rfcs/0000-windows-containers.md

```diff
@@ -0,0 +1,132 @@
+## Problem:
+
+kpack does not support building containers that can run on windows
+
+## Outcomes:
+
+### kpack should run on clusters that have windows workers
+
+kpack service pods and build pods for linux images will need to know not to run on the windows workers to work in a mixed-os cluster.
+
+The kpack release configuration should specify to run pods explicitly on linux os.
+
+The kpack controller should schedule linux builds on linux workers.
+
+**Actions to Take**
+
+Use the following node selector for all service pods and build pods for linux images:
+
+```
+nodeSelector:
+    kubernetes.io/os: linux
+```
+
+### kpack should restrict cluster stacks to be os-specific
+
+This would improve simplicity and usability. If you create mixed-os resources, it may be difficult to get out of the situation.
+
+If a user creates a cluster stack with build and run images of mixed os, the resource should be marked as not-ready
+
+**Actions to Take**
+
+- Add this validations to the kpack-controller
+
+Notes:
+- Stores will not be aware of the buildpackages os. There will be no constraint on creating stores with windows buildpackages and linux buildpackages.
+
+### kpack should support creating windows builders
+
+This is currently supported in the [pack cli](https://github.com/buildpacks/pack/issues/469)
+
+**Actions to Take**
+
+- kpack-controller will use a new environment variable `WINDOWS_LIFECYCLE_IMAGE` to reference the windows lifecycle image
+- Provide the kpack release (`release.yaml`) with a windows version of the lifecycle image and configure it with `WINDOWS_LIFECYCLE_IMAGE`
+- When a Builder or ClusterBuilder is created using a windows stack, kpack-controller must create a windows builder using the windows lifecycle image.
+    - Image os is determined by the image [os property](https://github.com/opencontainers/image-spec/blob/master/config.md#:~:text=os%20string).
+
+Notes:
+
+- Builders will not be aware of the buildpackages os. The only compatibility constraint between stacks and buildpackages for creating builders will be the stack id.
+
+### kpack should support running builds on windows workers
+
+**Actions to Take**
+
+- kpack-controller will accept new configuration via environment variable(s) for windows image(s) needed to run builds. The current behavior that must be recreated on windows is contained in the following images:
+    - `BUILD_INIT_IMAGE` - getting the build ready with creds and source
+    - `COMPLETION_IMAGE` - an image to run as a required regular (non-init) container
+    - `REBASE_IMAGE` - run a rebase
```

Comment:
As rebase does not need to execute anything specific to the windows operating system. We probably don't need a "windows" specific rebase image.